internal/controller/controllers/agentserviceconfig_controller_test.go

```diff
@@ -85,6 +87,212 @@ func AssertReconcileFailure(ctx context.Context, log logrus.FieldLogger, ascc AS
+			})
+		subject.Spec.OSImageCACertConfigMapRef = &corev1.LocalObjectReference{
+			Name: osImageCACertConfigMap.Name,
+		}
+		err := client.Create(ctx, osImageCACertConfigMap)
+		Expect(err).ToNot(HaveOccurred())
+		err = client.Create(ctx, subject)
+		Expect(err).ToNot(HaveOccurred())
+		awaitOSImageCACertValidationFail("expected to find single file `extra-ca.pem` in the ConfigMap")
+
+		// Now update the config map and see that the condition changes...
+		osImageCACertConfigMap = getOSImageCACertConfigMap(
+			map[string]string{
+				"extra-ca.pem": "-----BEGIN CERTIFICATE-----\nMIIDZTCCAk2gAwIBAgIUASRIJ1X9QHbJ/+daV+IjQdS1NIowDQYJKoZIhvcNAQEL\nBQAwQjELMAkGA1UEBhMCWFgxFTATBgNVBAcMDERlZmF1bHQgQ2l0eTEcMBoGA1UE\nCgwTRGVmYXVsdCBDb21wYW55IEx0ZDAeFw0yNDAxMDkxMzA0MzFaFw0zNDAxMDYx\nMzA0MzFaMEIxCzAJBgNVBAYTAlhYMRUwEwYDVQQHDAxEZWZhdWx0IENpdHkxHDAa\nBgNVBAoME0RlZmF1bHQgQ29tcGFueSBMdGQwggEiMA0GCSqGSIb3DQEBAQUAA4IB\nDwAwggEKAoIBAQC+Z9BqGKzPINWCSMdeTX52gLDoeTU3q4fH8QyHSO3hNo/eKtaE\nrHOqnsn/ntcsjFwX9Wfwxt1B73uqXkqWWCsH2QKGsw36gPJmSc6ZuqP7oUTApx0U\nOktdxOm96MouqN5OAXoPvzH5dFytJyW3TWpKJ3jP9ZWJrqmp4YcgnU+U6Vlen4iy\nN0NciJtdVDDsWoWqh0zg0YOHJpd43c7aQ0PFoPp4QEj4j29I7X91UmRP67dA8kSw\n2mPcZZFDkKY9fA0TuF1a3Dvx7yssvQoAC9F+jZYgBsTcFNGcc2roJVA8RwcdVZQ3\nbTwA0nLql5EDLdXHSthJiXHPhp6niOTsJx4bAgMBAAGjUzBRMB0GA1UdDgQWBBQr\nbklK4KlO6lgMM5MpVxqWcpWhxzAfBgNVHSMEGDAWgBQrbklK4KlO6lgMM5MpVxqW\ncpWhxzAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUAA4IBAQAEF3pv541h\nXKwDMqHShbpvqEqnhN74c6zc6b8nnIohRK5rNkEkIEf2ikJ6Pdik2te4IHEoA4V9\nHqtpKUtgNqge6GAw/p3kOB4C6eObZYZTaJ4ZiQ5UvO6R7w5MvFkjQH5hFO+fjhQv\n8whWWO7HRlt/Hll/VF3JNVALtIv2SGi51WHFqwe+ERKl0kGKWH8PyY4X6XflHQfa\n1FDev/NRnOVjRcXipsaZwRXcjRUiRX1KuOixlc8Reul8RdrL1Mt7lpl8+e/hqhoO\n2O5thhnTuV/mID3zE+5J8w6UcCdeZo4VDNWdZqPzrI/ymSgARVwUu0MFeYfCbMmB\neEpiSixb6YRM\n-----END CERTIFICATE-----\n",
+			})
+		err = client.Update(ctx, osImageCACertConfigMap)
+		Expect(err).ToNot(HaveOccurred())
+		awaitOSImageCACertValidationSuccess()
+
+		// Grab the statefulset and verify that a volume has been mapped
+		assertOSImageCACertVolumePresent()
+	})
+
+	It("Should create statful set with no OSImageCACert volume if OSImageCACert ConfigMap has not been supplied", func() {
```

Comment:
Do we have a test for the volume being created successfully when it is provided?

Reply:
Everything looks good except for this missing test case.

Reply:
I did actually add this, it's part of `assertOSImageCACertVolumePresent()`
I verify that there is a both a VolumeMount and a Volume mapped to that mount.

Is this what you meant?

Reply:
Ah sorry I missed the test case above this one. (`"Should add volume and volumemap for OSImageCACert when OSImageCACertRef is provided"`)